src/main/java/com/google/api/codegen/transformer/nodejs/NodeJSApiMethodParamTransformer.java

```diff
@@ -14,24 +14,165 @@
  */
 package com.google.api.codegen.transformer.nodejs;
 
+import com.google.api.codegen.config.MethodConfig;
 import com.google.api.codegen.transformer.ApiMethodParamTransformer;
 import com.google.api.codegen.transformer.MethodTransformerContext;
+import com.google.api.codegen.transformer.SurfaceNamer;
+import com.google.api.codegen.util.Name;
 import com.google.api.codegen.viewmodel.DynamicLangDefaultableParamView;
 import com.google.api.codegen.viewmodel.ParamDocView;
+import com.google.api.codegen.viewmodel.SimpleParamDocView;
+import com.google.api.tools.framework.model.Field;
+import com.google.api.tools.framework.model.TypeRef;
 import com.google.common.collect.ImmutableList;
 import java.util.List;
 
 public class NodeJSApiMethodParamTransformer implements ApiMethodParamTransformer {
   @Override
   public List<DynamicLangDefaultableParamView> generateMethodParams(
       MethodTransformerContext context) {
-    // TODO(eoogbe): implement this method when migrating to MVVM
-    return ImmutableList.<DynamicLangDefaultableParamView>of();
+    ImmutableList.Builder<DynamicLangDefaultableParamView> methodParams = ImmutableList.builder();
+    methodParams.addAll(generateDefaultableParams(context));
+
+    TypeRef arrayType = TypeRef.fromPrimitiveName("string").makeRepeated();
+
+    DynamicLangDefaultableParamView.Builder optionalArgs =
+        DynamicLangDefaultableParamView.newBuilder();
+    optionalArgs.name(context.getNamer().localVarName(Name.from("optional", "args")));
+    optionalArgs.defaultValue(context.getTypeTable().getZeroValueAndSaveNicknameFor(arrayType));
```

Comment:
This was an unused artifact from the php transformer I used a template. This is fixed.